### ISSUE TITLE
Mobile data connection is not active

### ISSUE THREAD

**Issue by IvanDimitrovCode** [pwittchen/ReactiveNetwork#307]
So basically i have this code:
```kotlin 
     ReactiveNetwork
                .observeNetworkConnectivity(this)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe {
                    when (it.state()) {
                        NetworkInfo.State.CONNECTED -> {
                            Log.d("CONNNECTION", isNetworkAvailable().toString())
                        }
                        NetworkInfo.State.DISCONNECTED -> {
                            Log.d("CONNNECTION", isNetworkAvailable().toString())
                        }
                        else -> {
                        }
                    }
                }
```

When i toggle the WiFi the observer is always called. But for some reason when i toggle the Mobile Data it is not. Further more i found this scenario: 
1. Have WiFi and MobileData ON
2. Turn off Mobile Data -> No event occurred
3. Turn off WiFi -> WiFi Disconnected occurs (as type NONE for some reason) + the Mobile Data event comes after it).
This is not reproducible each time its kind of random of the events that occur. (Tested it on pixel Xl (emulator) and my actual phone (Huawei p10 lite))
I don't know if this is relevant but  this is what my manifest has :     
```xml
    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```

**Comment by pwittchen** [comment]
Thanks for reporting this issue. I'll take a look on that.
What Android version are you using?

**Comment by IvanDimitrovCode** [comment]
Huawei -> 8, Pixel ->9. I managed to find a scenario which is always reproducible hope it helps.
1. WiFi and Mobile ON
2. Turn Mobile OFF -> Nothing happens
3. Turn Mobile ON-> Nothing happens
4. Turn WiFi OFF -> State.DISCONNECTED happens 
5. Turn WiFi ON-> State.CONNECTED happens 
6. Turn Mobile OFF->  State.CONNECTED happens
Hope this helps.
Also if this might help this method works for connectivity: 
 ```kotlin
       val connectivityManager = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val builder = NetworkRequest.Builder()

        connectivityManager.registerNetworkCallback(builder.build(), object : ConnectivityManager.NetworkCallback() {
            override fun onAvailable(network: Network?) {
                super.onAvailable(network)
                Log.d("CONNNECTION", "onAvailable")
            }

            override fun onLost(network: Network?) {
                super.onLost(network)
                Log.d("CONNNECTION", "onLost")
            }
        }) 
```
Let me know if i can help in some other way.

**Comment by pwittchen** [comment]
Thanks for providing these steps. What I can see now, you're **not notified** that you're disconnected from the network when you are connected to at least one network (WiFi or Mobile). The question is if returning `State.DISCONNECTED` while you're actually **connected** is the valid information. I think, behaviour which can be observed now is correct. To ensure this fact, we can try to reproduce the same thing using `BroadcastReceiver` or `NetworkCallback` from the Android SDK without RxJava and other additional libraries, but I suppose the result will be the same. I also don't want to change original behaviour of the Android SDK in this library.

**Comment by IvanDimitrovCode** [comment]
There is also the "Turn Mobile OFF-> State.CONNECTED happens" which is really incorrect.

**Comment by pwittchen** [comment]
Ok, I'll try to reproduce that.

**Comment by pwittchen** [comment]
This bug is fixed by PR #359.
Bug-fix will be available in the upcoming release.
Sorry that you had to wait for so long for it.
